### ISSUE TITLE
Add approximate mean, stddev and histogram/CDF QDigest UDFs

### ISSUE THREAD

**Issue by tanvach** [prestodb/presto#15009]
TLDR: Is it possible to add scalar UDFs that outputs approximate mean, stddev and histogram from Qdigest?

Hi, we've found that we can derive really useful approximate statistics base on the VALUES_AT_QUANTILES() function. In short, this function produces inverse CDF, and if sampled at uniform increments, [approximately resamples the data](https://en.wikipedia.org/wiki/Inverse_transform_sampling). For example:

```
SELECT
    AVG(value) AS mean,
    STDDEV(value) AS std_dev
FROM (
    SELECT
        VALUES_AT_QUANTILES(
            value_qdg,
            TRANSFORM(SEQUENCE(10, 990), (q) -> q * 0.001)
        ) AS value_array
    FROM your_data_table
)
CROSS JOIN UNNEST(value_array) t (value)
```

We can then extract the approximate CDF and histogram. Converting to CDF requires some array manipulation to make the x-axis spaced equally.

![97065336_157758452420200_2037842895393783808_o](https://user-images.githubusercontent.com/729913/89933223-9dc10600-dbc3-11ea-969b-3288780be961.png)

```
WITH invese_cdf_array AS (
 SELECT
    MAP(
       TRANSFORM(SEQUENCE(10, 990), (q) -> q * 0.001), 
       VALUES_AT_QUANTILES(
           value_qdg,
           TRANSFORM(SEQUENCE(10, 990), (q) -> q * 0.001)
       )
    ) AS inv_cdf
 FROM your_data_table
),
bin_edges_table AS (
 SELECT 
    CASE 
       WHEN x_bin_idx=30 THEN max_value+0.00000001*(max_value-min_value)/31 
       WHEN x_bin_idx=0 THEN min_value-0.00000001*(max_value-min_value)/31 
       ELSE min_value + x_bin_idx*(max_value-min_value)/31
    END AS x
 FROM (
    SELECT 
      ARRAY_MIN(MAP_VALUES(d.inv_CDF)) AS min_value,
      ARRAY_MAX(MAP_VALUES(d.inv_CDF)) AS max_value
    FROM invese_cdf_array d
 ) CROSS JOIN UNNEST(SEQUENCE(0, 30)) AS t(x_bin_idx)
),
actual_cdf_table AS (
 SELECT
   x,
   ARRAY_MAX(TRANSFORM(MAP_ENTRIES(d.inv_CDF), y -> IF(y[2] < x, y[1], 0))) AS y
 FROM inverse_cdf d
 CROSS JOIN bin_edges_table
)
```

As you can see, this is quite cumbersome, especially since QDigest data structure used in Presto actually has iterable and histogram interface that can compute these outputs efficiently.

**Comment by rongrong** [comment]
If you are proposing to add new functions, please add description of function signature and behavior. Thanks!
